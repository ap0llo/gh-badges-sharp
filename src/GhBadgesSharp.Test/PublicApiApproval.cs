using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using PublicApiGenerator;
using Xunit;

namespace Grynwald.GhBadgesSharp.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class PublicApiApproval
    {
        [Fact]
        public void Assembly_must_not_have_unapproved_API_changes()
        {
            // ARRANGE
            var assembly = typeof(Badge).Assembly;

            // ACT
            var publicApi = ApiGenerator.GeneratePublicApi(assembly);

            // ASSERT
            var writer = new ApprovalTextWriter(publicApi);
            Approvals.Verify(writer, new UnitTestFrameworkNamer(), Approvals.GetReporter());
        }
    }
}
