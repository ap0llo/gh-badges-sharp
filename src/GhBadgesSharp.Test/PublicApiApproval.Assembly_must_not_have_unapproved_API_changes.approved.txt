[assembly: System.Runtime.CompilerServices.InternalsVisibleToAttribute("GhBadgesSharp.Test")]
[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETStandard,Version=v2.0", FrameworkDisplayName="")]
namespace GhBadgesSharp
{
    public class static Badge
    {
        public static System.Xml.Linq.XElement MakeBadge(GhBadgesSharp.BadgeStyle style, string leftText, string rightText, string color = null, string labelColor = null, string leftLink = null, string rightLink = null) { }
    }
    public enum BadgeStyle
    {
        Flat = 0,
        FlatSquare = 1,
        Plastic = 2,
        ForTheBadge = 3,
        Social = 4,
    }
}